lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1020,7 +1022,9 @@ public function recomputeSingleEntityChangeSet(ClassMetadata $class, $entity)
         foreach ($class->reflFields as $name => $refProp) {
             if (( ! $class->isIdentifier($name) || ! $class->isIdGeneratorIdentity())
                 && ($name !== $class->versionField)
-                && ! $class->isCollectionValuedAssociation($name)) {
+                && ! $class->isCollectionValuedAssociation($name)
+                && (method_exists($refProp, 'isInitialized') && $refProp->isInitialized($entity))
```

Comment:
Probably best to have a check on whether it's an uninitialized proxy or not: an entity with uninitialized fields landing in the `UnitOfWork` is very very likely a bug...

Reply:
Looks like it. I noticed that this PR might also be redundant and is missing some code, so I am closing it for now.
Thanks for having a look!